tap_klaviyo/utils.py

```diff
@@ -3,13 +3,70 @@
 import singer
 from singer import metrics
 import requests
+import backoff
+import json
+import simplejson
 
 DATETIME_FMT = "%Y-%m-%dT%H:%M:%SZ"
 
-
 session = requests.Session()
 logger = singer.get_logger()
 
+class KlaviyoError(Exception):
+    pass
+
+class KlaviyoNotFoundError(KlaviyoError):
+    pass
+
+class KlaviyoBadRequestError(KlaviyoError):
+    pass
+
+class KlaviyoUnauthorizedError(KlaviyoError):
+    pass
+
+class KlaviyoForbiddenError(KlaviyoError):
+    pass
+
+class KlaviyoInternalServiceError(KlaviyoError):
+    pass
+
+ERROR_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "raise_exception": KlaviyoBadRequestError,
+        "message": "Request is missing or has a bad parameter."
+    },
+    401: {
+        "raise_exception": KlaviyoUnauthorizedError,
+        "message": "Invalid authorization credentials."
+    },
+    403: {
+        "raise_exception": KlaviyoForbiddenError,
+        "message": "Request is missing or has an invalid API key."
+    },
+    404: {
+        "raise_exception": KlaviyoNotFoundError,
+        "message": "The requested resource doesn't exist."
+    },
+    500: {
+        "raise_exception": KlaviyoInternalServiceError,
+        "message": "Something is wrong on Klaviyo's end."
```

Comment:
Can we also adjust this to say "Internal Service Error from Klaviyo"

Reply:
Added the string in message